share/qtcreator/sfdk/modules/20-building-mb2/doc/module.adoc

```diff
@@ -85,15 +85,31 @@ Similarly it is possible to supply any custom build-time dependencies by simply
 
 === Clean Builds
 
-Build time requirements of one package may interfere with those of another one. In order to avoid polluting build targets with dependencies of a particular package, sfdk uses build targets through writable snapshots. See the description of the 'snapshot' configuration option for details on how a snapshot is chosen.
+Build time requirements of one package may interfere with those of another one. Packages modified and deployed under the build environment in scope of one task may not be compatible with changes worked on in scope of other task. As time goes the build environment accumulates various changes, leading to generally non-reproducible build results.
```

Comment:
I think it could be mentioned, preferably quite early, that most of the time developer does not have to think about the snapshots or how they work, as long as all the build time requirements are defined in the RPM 'SPEC' file.

Reply:
I was thinking how/where to fit such a note but have not found a feasible way. Also, it should not be needed. What are the issues with missing BuildRequires should be something the user learns very soon, both from the preceding parts of this manual and from their own experience. It is more likely to make troubles to power users or new users advised by old power users to manually deliver locally built dependencies, which is covered by

>  Use a shared output directory to supply locally built build-time dependencies without disturbance by (possibly implied) 'build-requires reset'. See WORKING ON MULTIPLE PACKAGES above.

It also speak in terms of 'build-requires reset', as snapshots are intentionally mentioned only below the "Detailed information follows." line.